docker/files/run/run.sh

```diff
@@ -111,18 +111,19 @@ function start_private_network() {
   apply_configuration
 
   # TODO: Is there a way to properly exec a private network?
-  goal network start -r "$ALGORAND_DATA/.."
-  tail -f "$ALGORAND_DATA/node.log"
+  goal network start -r "${ALGORAND_DATA}/.."
+  tail -f "${ALGORAND_DATA}/node.log"
 }
 
 function start_new_private_network() {
-  cd /node
+  cd /algod
   local TEMPLATE="template.json"
   if [ "$DEV_MODE" ]; then
     TEMPLATE="devmode_template.json"
   fi
   sed -i "s/NUM_ROUNDS/${NUM_ROUNDS:-30000}/" "run/$TEMPLATE"
-  goal network create -n dockernet -r "$ALGORAND_DATA/.." -t "run/$TEMPLATE"
+  goal network create --noclean -n dockernet -r "/tmp/dockernet" -t "run/$TEMPLATE"
+  mv -v /tmp/dockernet/* "${ALGORAND_DATA}/.."
```

Comment:
Why two steps here instead of letting goal network create put files in the correct location?

Reply:
That warning is a feature not a bug. If the directory is not suitable for a private network it's important to understand why. At times I was careful to create the network before updating settings to avoid this error. We also updated "goal network create" to allow empty directories, are there other specific exceptions we should add?

Reply:
Nice sleuthing. That `IsEmpty` line was added for the docker container so that you could mount an empty directory: https://github.com/algorand/go-algorand/pull/3911

Maybe changing the config file override directory would resolve the problems you saw?